awriter/signer.go

```diff
@@ -61,18 +60,9 @@ func SignExisting(src io.Reader, dst io.Writer, key []byte, overwrite bool) erro
 			return errors.Wrap(err, "Could not write tar header")
 		}
 
-		read, err := rTar.Read(buf)
-		if err != nil && err != io.EOF {
-			return errors.Wrap(err, "Could not read tar file entry")
-		} else if read == 0 {
-			continue
-		}
-
-		written, err := wTar.Write(buf[:read])
+		_, err = io.Copy(wTar, rTar)
```

Comment:
Better to use the io.Copy functionality than basically juggle readers and writers ourself imo.

Reply:
:+1: